### PR TITLE
rewrite plugin to work with drone 0.8+

### PR DESCRIPTION
**Pull request by xoxys**
I picket up the work from https://github.com/drone-plugins/drone-pypi/pull/15 , so changes are nearly the same:
- Removed .drone.sec - This is no longer valid
- Updated Dockerfile - install Python3 rather than python2 and add extra label-schema labels
- Switched to Drone 0.8 style plugin.go and main.go (this also means deleting main_test.go and
- switching to plugin_test.go
- Removed the Maintainers file - should maybe keep this
- switch to twine for pypi uploading

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -1,38 +1,60 @@
...
-    - CGO_ENABLED=0
-    - DRONE_PYPI_PATH=testdata
-    - DRONE_PYPI_REPOSITORY=http://localhost:8000
-    - DRONE_PYPI_DISTRIBUTIONS=sdist
-  commands:
-    - apt-get update && apt-get install -y python-setuptools
-    - go get
-    - go build
-    - go test -cover
+    - PLUGIN_REPOSITORY=https://test.pypi.org/legacy/
```
Comment (tboerger) [inline .drone.yml]: Can't we just start a simple pypi server locally as part of the service definition? Presse write the environment block before the commands block :)

File: Dockerfile [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
-	py-pip \
-	python \
- && rm -rf /var/cache/apk/* \
- && pip install --no-cache-dir --upgrade \
-	pip \
-	setuptools \
-	wheel
+ARG GITSHA="VCS ref not found"
+ARG BUILDDATE="Build date not found"
+LABEL org.label-schema.url="https://github.com/drone-plugins/drone-pypi" \
```
Comment (tboerger) [inline Dockerfile]: Please take the Dockerfile from the codecov plugin as an example. Multiple of the label values are already set by drone

File: README.md [hunk 3]
```diff
@@ -1,68 +1,28 @@
...
-CGO_ENABLED=0 go build -a -tags netgo
-docker build --rm=true -t plugins/drone-pypi .
+```shell
+docker run --rm \
+  -e PLUGIN_USERNAME=jdoe \
+  -e PLUGIN_PASSWORD=my_secret \
+  -e PLUGIN_SKIP_BUILD=false \
+  -v $(pwd):$(pwd) \
+  -w $(pwd) \
+  plugins/drone-pypi
```
Comment (tboerger) [inline README.md]: plugins/pypi

File: main.go [hunk 4]
```diff
@@ -1,122 +1,74 @@
...
-	"io"
-	"log"
 	"os"
-	"os/exec"
-	"path"
-	"strings"
 
-	"github.com/drone/drone-go/drone"
-	"github.com/drone/drone-go/plugin"
+	"github.com/codegangsta/cli"
```
Comment (tboerger) [inline main.go]: urfave/cli should be correct today

File: .drone.yml (hunk 1 above)
Comment (xoxys) [reply .drone.yml]: Good idea, definitely possible. Will do this after Christmas. Merry Christmas to all of you also.

File: Dockerfile (hunk 2 above)
Comment (xoxys) [reply Dockerfile]: Done

File: README.md (hunk 3 above)
Comment (xoxys) [reply README.md]: Done

File: main.go (hunk 4 above)
Comment (xoxys) [reply main.go]: Done

File: Dockerfile (hunk 2 above)
Comment (xoxys) [reply Dockerfile]: What about plugins/base image maybe switch to this one?

File: Dockerfile [hunk 5]
```diff
@@ -1,15 +1,13 @@
...
-	ca-certificates \
-	py-pip \
-	python \
- && rm -rf /var/cache/apk/* \
- && pip install --no-cache-dir --upgrade \
-	pip \
-	setuptools \
-	wheel
+LABEL maintainer="Drone.IO Community <drone-dev@googlegroups.com>" \
+  org.label-schema.name="Drone Codecov" \
```
Comment (tboerger) [inline Dockerfile]: Drone PyPi :)

File: Dockerfile (hunk 2 above)
Comment (tboerger) [reply Dockerfile]: Yeah, makes sense. This base image will be updated pretty soon.

**Comment by xoxys** [comment]
@tboerger Got it working and do also some refactoring to match the new plugin base structure :) Windows build is current not included, I think I need some base introducing...

I needed much commits to get it running finally, so maybe we should merge it with `squash`

File: .drone.yml (hunk 1 above)
Comment (xoxys) [reply .drone.yml]: Done

File: Dockerfile (hunk 2 above)
Comment (xoxys) [reply Dockerfile]: Done

File: Dockerfile (hunk 5 above)
Comment (xoxys) [reply Dockerfile]: Also done :)

**Comment by xoxys** [comment]
Rebased to cleanup history

**Comment by xoxys** [comment]
Yeah, there were some more stupid copy paste errors... I've grepped over the project and fixed the parts.

File: .github/settings.yml [hunk 6]
```diff
@@ -0,0 +1,75 @@
+repository:
+  name: drone-ansible
```
[the remark this answers is not in the document]
Comment (xoxys) [reply .github/settings.yml]: Done

File: docker/Dockerfile.linux.amd64 [hunk 7]
```diff
@@ -0,0 +1,13 @@
+FROM plugins/base:linux-amd64
+
+LABEL maintainer="Drone.IO Community <drone-dev@googlegroups.com>" \
+  org.label-schema.name="Drone Ansible" \
```
[the remark this answers is not in the document]
Comment (xoxys) [reply docker/Dockerfile.linux.amd64]: Done

File: go.mod [hunk 8]
```diff
@@ -0,0 +1,7 @@
+module github.com/xoxys/plugin-pypi
+
+require (
+	github.com/codegangsta/cli v1.20.0
```
[the remark this answers is not in the document]
Comment (xoxys) [reply go.mod]: Done

**Comment by xoxys** [comment]
Certainly :) give me a second

**Pull request merged**